includes/Modules/TagManager.php

```diff
@@ -459,18 +485,28 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 					} else {
 						$account_id = $response['accounts'][0]->getAccountId();
 					}
+
 					$containers = $this->get_data( 'list-containers', array( 'accountId' => $account_id ) );
+
 					if ( is_wp_error( $containers ) ) {
 						return $response;
 					}
+
 					return array_merge( $response, $containers );
 				case 'list-containers':
-					$response = array(
+					$account_id = $response['accountId'];
```

Comment:
I see we need to reuse the passed `accountId` here where we don't have access to the originally passed value. This is an internal shortcoming that needs to be resolved at some point. However, this shortcoming should be worked around _internally_, not _externally_ by modifying the response to account for it. While seemingly even more hacky, please work around this by temporarily setting an internal property so that it can be reaccessed afterwards - this will leave the API consumer unaffected. See https://github.com/google/site-kit-wp/pull/335/files#diff-b2484d51f9cd2114eca74b872d92b73eR45 for related code that already works around a similar issue.

Reply:
Thanks @felixarntz, I did think about this other option but changing the response seemed somehow a bit less hacky, I have changed it here https://github.com/google/site-kit-wp/pull/335/commits/55f1464b45b3f142227eca05b8a65d39495fd267